### ISSUE TITLE
False positive detection of CV05

### ISSUE THREAD

**Issue by bujocek** [sqlfluff/sqlfluff#7116]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Sqlfluff lint in [VSCode sqlfluff extension](https://marketplace.visualstudio.com/items?itemName=dorzey.vscode-sqlfluff) incorrectly marks variable assignments in SELECT statemens as NULL comparison.

<img width="1532" height="234" alt="Image" src="https://github.com/user-attachments/assets/5e4accd6-6723-4c91-b0bd-31cff930a4f8" />

Might be related to #6000 

### Expected Behaviour

It should warn about [CV05](https://docs.sqlfluff.com/en/stable/reference/rules.html#sqlfluff.rules.sphinx.Rule_CV05) only for comparison (i.e. in WHERE or CASE parts of the statements) but not for assignments

### Observed Behaviour

This is my result from console for the example code in "How to reproduce" block

```
--------------------Executing Command--------------------

sqlfluff lint --format json --config C:/Develop/database-scripts/.sqlfluff --dialect tsql --stdin-filename ../../Users/jonas.bujok/AppData/Local/Programs/Microsoft VS Code/Untitled-3 -

------------------------------------------------------------

Received close event, code 1 signal null
Raw stdout output:

------------------------------------------------------------

[{"filepath": "stdin", "violations": [{"start_line_no": 3, "start_line_pos": 1, "code": "LT09", "description": "Select targets should be on a new line unless there is only one select target.", "name": "layout.select_targets", "warning": true, "fixes": [{"type": "replace", "edit": " @MyVariable = NULL", "start_line_no": 3, "start_line_pos": 7, "start_file_pos": 40, "end_line_no": 4, "end_line_pos": 1, "end_file_pos": 41}, {"type": "delete", "edit": "", "start_line_no": 4, "start_line_pos": 3, "start_file_pos": 43, "end_line_no": 4, "end_line_pos": 21, "end_file_pos": 61}, {"type": "delete", "edit": "", "start_line_no": 4, "start_line_pos": 1, "start_file_pos": 41, "end_line_no": 4, "end_line_pos": 3, "end_file_pos": 43}, {"type": "create_after", "edit": "\n  ", "start_line_no": 4, "start_line_pos": 21, "start_file_pos": 61, "end_line_no": 4, "end_line_pos": 21, "end_file_pos": 61}], "start_file_pos": 34, "end_line_no": 4, "end_line_pos": 21, "end_file_pos": 61}, {"start_line_no": 4, "start_line_pos": 15, "code": "CV05", "description": "Comparisons with NULL should use \"IS\" or \"IS NOT\".", "name": "convention.is_null", "warning": false, "fixes": [{"type": "replace", "edit": "IS", "start_line_no": 4, "start_line_pos": 15, "start_file_pos": 55, "end_line_no": 4, "end_line_pos": 16, "end_file_pos": 56}], "start_file_pos": 55, "end_line_no": 4, "end_line_pos": 16, "end_file_pos": 56}], "statistics": {"source_chars": 63, "templated_chars": 63, "segments": 42, "raw_segments": 28}, "timings": {"templating": 0.0, "lexing": 0.0, "parsing": 0.014999999897554517, "linting": 0.0, "AL01": 0.0, "AL02": 0.0, "AL04": 0.0, "AL05": 0.0, "AL06": 0.0, "AL07": 0.0, "AL08": 0.0, "AL09": 0.0, "AM01": 0.0, "AM02": 0.0, "AM03": 0.0, "AM04": 0.0, "AM06": 0.0, "AM07": 0.0, "AM08": 0.0, "CP01": 0.0, "CP03": 0.0, "CP04": 0.0, "CV01": 0.0, "CV02": 0.0, "CV03": 0.0, "CV04": 0.0, "CV05": 0.0, "CV06": 0.0, "CV07": 0.0, "CV08": 0.0, "CV09": 0.0, "CV10": 0.0, "CV11": 0.0, "CV12": 0.0, "JJ01": 0.0, "LT01": 0.0, "LT03": 0.0, "LT04": 0.0, "LT06": 0.0, "LT07": 0.0, "LT09": 0.0, "LT10": 0.0, "LT11": 0.0, "LT12": 0.0, "LT13": 0.0, "LT14": 0.0, "LT15": 0.0, "RF01": 0.0, "RF02": 0.0, "RF03": 0.0, "RF04": 0.0, "RF05": 0.0, "RF06": 0.0, "ST01": 0.0, "ST02": 0.0, "ST03": 0.0, "ST04": 0.0, "ST05": 0.0, "ST07": 0.0, "ST08": 0.0, "ST09": 0.0, "ST10": 0.0, "ST11": 0.0, "TQ01": 0.0}}]

------------------------------------------------------------
```

### How to reproduce

example code:
```
DECLARE @MyVariable VARCHAR(10);

SELECT
  @MyVariable = NULL;
```

### Dialect

tsql

### Version

3.4.2

### Configuration

settings json part from my VSCode  plugin:
```
    "sqlfluff.rules": [],
    "sqlfluff.codeActions.noqa": [
    ],
    "sqlfluff.experimental.format.executeInTerminal": true,
    "sqlfluff.config": "${workspaceFolder}/.sqlfluff",
    "sqlfluff.dialect": "tsql"
```

.sqlfluff in my project:
```
[sqlfluff]
dialect = tsql
exclude_rules = LT05, CP05, CP02, AL03, LT02, AM05, L022, ST06
warnings = LT01, LT09, RF05
large_file_skip_byte_limit = 30000

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
indented_joins = True

[sqlfluff:rules:structure.join_condition_order]
preferred_first_table_in_join_clause = later
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
